real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/LogicErrorMessageConsole.java

```diff
@@ -0,0 +1,173 @@
+package com.roboclub.robobuggy.ui;
+
+import java.awt.Color;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+import javax.swing.JButton;
+import javax.swing.JLabel;
+import javax.swing.JTextArea;
+
+import com.roboclub.robobuggy.main.RobobuggyMessageLevel;
+import com.roboclub.robobuggy.messages.RobobuggyLogicExceptionMeasurment;
+import com.roboclub.robobuggy.ros.Message;
+import com.roboclub.robobuggy.ros.MessageListener;
+import com.roboclub.robobuggy.ros.NodeChannel;
+import com.roboclub.robobuggy.ros.Subscriber;
+
+public class LogicErrorMessageConsole extends RoboBuggyGUIContainer{
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = -6532100079403054035L;
+	static final RobobuggyMessageLevel[] robobuggyMessageLevels = RobobuggyMessageLevel.values();
+	static final int NUMBER_OF_MESSAGES = robobuggyMessageLevels.length;
+	boolean[] show_messages;
+	JButton[] buttons;
+	JTextArea messages;
+	
+	String messageBuffer[] = new String[10];
+	int messageBufferStart = 0;
+	int messageBufferEnd = 0;
+
+	JButton EXCEPTION_btn;
+	JButton WARNING_btn;
+	JButton NOTE_btn;
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1760190883/issues/source?bid=2938665&fileBranchId=3059924#l35)